statePattern.py

```diff
@@ -0,0 +1,115 @@
+import abc
+import defs
+
+# State machine implemented with state pattern
+
+
+class Context:
+
+    def __init__(self):
+        self.state = Off()
+
+    def work(self, pool):
+        self.state.handle(self, pool)
+
+
+class State(metaclass=abc.ABCMeta):
+
+    @abc.abstractmethod
+    def get_name(self):
+        pass
+
+    @abc.abstractmethod
+    def handle(self, context, pool):
+        pass
+
+
+class On(State):
+
+    def get_name(self):
+        return defs.ON
+
+    def handle(self, context, pool):
+        pool.handle_valve()
+
+        if pool.get_next_state() == defs.FOFF:
+            pool.set_next_state("")
+            pool.set_state(defs.FOFF)
+            context.state = Foff()
+            return
+
+        if pool.floatSwitch.get_state() == 0:
```

Comment:
Would it read better if all the `pool.floatSwitch.getState == 0` && `pool.floatSwitch.getState == 1` would be rewritten as:

```
def isFloatSwitchOn(pool): return pool.floatSwitch.get_state() == 1
...
if !isFloatSwitchOn(pool):
  ...
elif isFloatSwitchOn(pool):
  ...
```
?

Reply:
Good idea.